### PR TITLE
win-py27

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

~~Just trying out stuff out of curiosity.~~

See https://github.com/conda-forge/llvmlite-feedstock/pull/26#issuecomment-571542245:
> > At conda-forge, we are only using VS2008 for python 2.7 extensions. Why is this used here?
> 
> 1. We don't build LLVM with VS2008.
> 2. This restores behavior from previous builds: https://github.com/conda-forge/llvmlite-feedstock/blob/1ce3355ede77338761c56db5eec94002e1c470a0/recipe/meta.yaml#L39
> 3. If we don't do it here, users will get the builds from Anaconda which do the same VS2015 injection.
> 
> This does not add anything *new* but just puts back the status quo into `conda-forge` again.
> Also, Numba is deprecating 2.7 support so we (and Anaconda/`defaults`) can remove this soon, hopefully.

### CODE REVIEW

**Comment by henryiii** [comment]
Is this working?

**Comment by isuruf** [comment]
At conda-forge, we are only using VS2008 for python 2.7 extensions. Why is this used here?

**Comment by henryiii** [comment]
Because the last 2.7 release of Numba (0.47) can't be built on Windows because llvmlite is missing. In the previous version, it was pulling in llvmlite from defaults from what I've been told, which does include a 2.7 build on Windows.

I don't use 2.7 or Windows personally, I was just was trying not to loose a platform when upgrading Numba.

**Comment by isuruf** [comment]
Using VS2017 on python 2.7 extensions would require a CFEP.

**Comment by henryiii** [comment]
I doubt that would go very far, as I don't think supporting 2.7, especially on Windows, would get much Conda support. But this does mean that Numba will join the list of packages unsupported on conda-forge on Windows for 2.7. Though they are doing this in defaults.

For now, I think the idea is to push the (Numba) upgrade with 2.7 + vs<14 skipped, then eventually it would be added if someone cares enough (to write and push a CFEP, apparently).

**Comment by mbargull** [comment]
> At conda-forge, we are only using VS2008 for python 2.7 extensions. Why is this used here?

1. We don't build LLVM with VS2008.
2. This restores behavior from previous builds: https://github.com/conda-forge/llvmlite-feedstock/blob/1ce3355ede77338761c56db5eec94002e1c470a0/recipe/meta.yaml#L39
3. If we don't do it here, users will get the builds from Anaconda which do the same VS2015 injection.

This does not add anything *new* but just puts back the status quo into `conda-forge` again.
Also, Numba is deprecating 2.7 support so we (and Anaconda/`defaults`) can remove this soon, hopefully.

> Using VS2017 on python 2.7 extensions would require a CFEP.

I have absolutely no intention to push for that and hope no on else is doing so. If some big corp sees the need to burn some money on this, let 'em have it and deal with the ABI and incompatible eggs/wheels. But I see no reason why we should support this.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,9 +19,7 @@ requirements:
   build:
     - {{ compiler('c') }}    # [unix or (win and py==27)]
     - {{ compiler('cxx') }}  # [unix or (win and py==27)]
-    - vs2017_win-64          # [win64 and py>27]
-    - vs2017_win-32          # [win32 and py>27]
-
+    - vs2017_{{ target_platform }}  # [win and py>27]
```
Comment (mbargull) [inline recipe/meta.yaml]: Rerendering with this lead to `contains(variables['CONFIG'], 'vs2008')` in `.azure-pipelines/azure-pipelines-win.yml` not being triggered.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,9 +29,11 @@ requirements:
     - enum34          # [py27]
     # llvmdev is built with libz compression support
     - zlib            # [unix]
+    - vs2015_runtime  # [win and py==27]
```
Comment (isuruf) [inline recipe/meta.yaml]: This needs a lower bound

File: recipe/meta.yaml (hunk 1 above)
Comment (mbargull) [reply recipe/meta.yaml]: Which means we should either go with
```yaml
    - vs2017_win-64          # [win64 and py>27]
    - vs2017_win-32          # [win32 and py>27]
```
from https://github.com/conda-forge/llvmlite-feedstock/pull/26/commits/36d405f683845ef64e081063fc5891e0f827b107
or the `conda_build_config.yaml`
from https://github.com/conda-forge/llvmlite-feedstock/pull/26/commits/eca1de4e9baa595d6e8769611e50dd749375946a

File: recipe/meta.yaml (hunk 2 above)
Comment (mbargull) [reply recipe/meta.yaml]: As in `pin_compatible` or something specific?

Comment (isuruf) [reply recipe/meta.yaml]: You need the one from vc 14.1 instead of from vc 14

**Comment by isuruf** [comment]
My experience has been that when we allow such things, then everybody starts using it. That's why I said you need a CFEP. cc @conda-forge/core

File: recipe/meta.yaml (hunk 2 above)
Comment (mbargull) [reply recipe/meta.yaml]: Ah, makes sense, thanks.

**Comment by mbargull** [comment]
> My experience has been that when we allow such things, then everybody starts using it.

Happy to add a comment to the `vs2015_runtime  # [win and py==27]` lines saying not to use this hack. Any phrasing suggestions?

> That's why I said you need a CFEP.

Proposing what then? I don't want use to have a document saying "you can do this, it's okay in some cases". Ideally, this is the only case and in a couple of weeks/months we can remove this again. (If you want to, we can also just remove it right away from the recipe when a build is out.)

**Comment by henryiii** [comment]
Is this moving forward with a decision? It's blocking the Numba upgrade and therefore the Python 3.8 migration ATM.

**Comment by jakirkham** [comment]
Why do you see this as blocking? We don’t build Numba for Python 2.7 on Windows currently.

**Comment by mbargull** [comment]
We *do build* Numba for Python 2.7 on Windows currently. With that PR we *would stop* doing so.

I really would like us to put out one last round of 2.7 builds for `llvmlite 0.31.0` and `numba 0.47.0` and then be done with it.

TBH -- although I do highly value @isuruf's input/work generally -- I'm inclined to just hit "merge", despite his concerns on this. I don't see a high chance of this becoming a bandwagon for VS2017+Py2.7 since it will be a temporary measure and is happening via `defaults` anyway.

**Comment by jakirkham** [comment]
Huh. I guess you are right. Missed that.

That said, @isuruf makes a good point. In fact, we've also actually [already decided on this issue as an org]( https://github.com/conda-forge/conda-forge.github.io/issues/64 ) and [have not allowed others to do similar workarounds]( https://github.com/conda-forge/feather-format-feedstock/pull/7#issuecomment-248565760 ). I don't think it would be fair to others if we decided to special case things here.

I don't think we can pursue this unfortunately. Though I understand this is important to you. Perhaps we can work with `defaults` to update their package? That appears to be what we have used for Python 2.7 builds of `llvmlite` until now. WDYT?

**Comment by mbargull** [comment]
Well, I'm interested in a *technical* solution, not in a *political* one.
Asking Anaconda to put out a build so we can use that for `numba` is *purely political* since that solution would be *technically identical* to us uploading it.
If we wanted to be ultra strict, we'd even have to pull existing builds for `llvmlite>=0.23.0,<=0.26.0` (uploaded 2018, i.e., well after the discussions you referenced) so that people only get those from `defaults`.

It isn't really important to me, personally. The only interest from my part is because I just wouldn't want to concern myself with people asking for a `numba[subdir=win-64,version=0.47.0,build=*py27*]` due to https://github.com/conda-forge/numba-feedstock/pull/41#issuecomment-571134492:
> \[...\] 2. But there might be increased demand for the "last Numba version that supports Python 2.7".

**Comment by mbargull** [comment]
FYI, `llvmlite[channel=defaults,subdir=win-64,version=0.31.0,build=*py27*]` is up: https://anaconda.org/main/llvmlite/files?version=0.31.0
Meaning, merging this PR has little to no user impact.

@conda-forge/core, please merge or close this PR.

**Pull request closed without merging**